xwiki-rendering-api/src/main/java/org/xwiki/rendering/syntax/SyntaxType.java

```diff
@@ -123,6 +125,11 @@
      */
     public static final SyntaxType HTML = register("html", "HTML");
 
+    /**
+     * All HTML types.
+     */
+    public static final Set<SyntaxType> HTML_TYPES = SetUtils.hashSet(XHTML, HTML, ANNOTATED_XHTML, ANNOTATED_HTML);
```

Comment:
Note that there's another place registering a single component for all HTML syntax types and it needs to be refactored and moved into several components, each brought by the module which contributes its syntax.

Reply:
See https://jira.xwiki.org/browse/XRENDERING-608?focusedCommentId=108926&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-108926

Reply:
Said differently, if an extension/module contributing a new syntax is not present, the syntax and its syntax type (if it introduces a new syntax type) should not be present in the wiki.

Reply:
Note: In the case of XHTMLChainingRenderer what the code really wants to check is if the syntax can be directly injected into the returned browser content (i.e. if the browser supports the syntax).

Reply:
Regarding Annotated HTML, I think that ideally Syntax Types should be able to inherit a parent type. That would allow to register it under a type that inherit from the HTML Syntax Type. Same for Annotated XHTML.

Reply:
IMO, we can start with what @oanalavinia did here which is already useful and does not really do much harm and discuss the more long-term and generic solution we want in a less rushed manner.

By the way @oanalavinia don't forget the `@since` for this new API.

Reply:
> For the last option, with isBrowserSupported(), I would need to use the predefined syntaxes, right? Isn't it also against the new direction?

Not sure I understand. You want to know if a passed Syntax is a browser-supported syntax, right? So you'd just need to call `isBrowserSupported()`.

Reply:
So for now I will keep these changes (adding also the `@since` tag) and I'll open a different issue for solving this after in a better and correct manner.